components/src/core/IconSwitch/IconSwitch.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { StyleSheet, TouchableOpacity, ViewProps, View } from 'react-native';
+import Animated, {
+    interpolate,
+    useAnimatedStyle,
+    useSharedValue,
+    withTiming,
+    Easing,
+    Extrapolation,
+    interpolateColor,
+} from 'react-native-reanimated';
+import { Icon } from '../Icon/Icon';
+import { $DeepPartial } from '@callstack/react-theme-provider';
+import { MD3Theme, useTheme } from 'react-native-paper';
+import Color from 'color';
+
+export type IconSwitchProps = ViewProps & {
+    showIcon?: boolean;
+    disabled?: boolean;
+    value?: boolean;
+    onValueChange?: (arg: boolean) => void;
+    theme?: $DeepPartial<MD3Theme>;
+};
+
+export const IconSwitch: React.FC<IconSwitchProps> = (props) => {
+    const { showIcon = false, disabled = false, value = false, onValueChange } = props;
+    const theme = useTheme(props.theme);
+
+    const [toggled, setToggled] = useState(value);
+    const shareValue = useSharedValue(value ? 1 : 0);
+
+    const containerScale = {
+        height: 32,
+        width: 52,
+    };
+    const switchScale = {
+        width: toggled ? 24 : showIcon ? 24 : 16,
+        height: toggled ? 24 : showIcon ? 24 : 16,
+    };
+
+    const onChangeToggle = (): void => {
+        setToggled(!toggled);
+        onValueChange?.(!toggled);
+    };
+
+    const onPressSwitch = (): void => {
+        if (shareValue.value === 0) {
+            shareValue.value = withTiming(1, {
+                duration: 100,
+                easing: Easing.bezier(0.4, 0.0, 0.2, 1),
+            });
+        } else {
+            shareValue.value = withTiming(0, {
+                duration: 100,
+                easing: Easing.bezier(0.4, 0.0, 0.2, 1),
+            });
+        }
+        onChangeToggle();
+    };
+
+    const styles = StyleSheet.create({
+        containerStyle: {
+            display: 'flex',
+            justifyContent: 'center',
+            backgroundColor: disabled
+                ? Color('#BDCAD1').alpha(0.3).rgb().string()
+                : toggled
+                ? theme.colors.primary
+                : // @ts-ignore
+                  theme.colors.surfaceContainerHighest,
+            // @ts-ignore
+            borderColor: toggled ? undefined : disabled ? theme.colors.disabled : theme.colors.outline,
+            borderWidth: toggled ? 0 : 2,
+            borderRadius: 100,
+        },
+        switchButton: {
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+            borderRadius: 23,
+            marginLeft: 4,
```

Comment:
The left spacing does not look correct to me
![image](https://github.com/etn-ccis/blui-react-native-component-library/assets/8997218/010c007a-6965-4e25-b9f9-ed0a1ea01dde)